### PR TITLE
MAX_CHANNELS as board specific

### PR DESCRIPTION
**Pull request by pmiettinen**

### CODE REVIEW

File: src/board/v2/Makefile.inc [hunk 1]
```diff
@@ -11,6 +11,11 @@ ifeq ($(USE_PROCESS_STACKSIZE),)
   USE_PROCESS_STACKSIZE = 0x738
 endif
 
+# Max tracking channels.
+ifeq ($(MAX_CHANNELS),)
+  MAX_CHANNELS = 11
```
Comment (adel-mamin) [inline src/board/v2/Makefile.inc]: I would suggest to use (MAX_CHANNELS + 1) to define NAP_MAX_N_TRACK_CHANNELS value for Piksi v2 and v3.

File: src/board/v2/nap/nap_hw.h [hunk 2]
```diff
@@ -39,7 +39,7 @@
 #define NAP_HASH_NOTREADY           2
 
 /** Max number of tracking channels NAP configuration will be built with. */
-#define NAP_MAX_N_TRACK_CHANNELS     12
+#define NAP_MAX_N_TRACK_CHANNELS    MAX_CHANNELS
```
Comment (adel-mamin) [inline src/board/v2/nap/nap_hw.h]: Shouldn't it be (MAX_CHANNELS + 1) ?

File: src/board/v3/Makefile.inc [hunk 3]
```diff
@@ -46,6 +46,11 @@ ifeq ($(USE_FPU),)
   USE_FPU = hard
 endif
 
+# Max tracking channels.
+ifeq ($(MAX_CHANNELS),)
+  MAX_CHANNELS = 31
```
Comment (adel-mamin) [inline src/board/v3/Makefile.inc]: Same here.

File: src/board/v3/nap/nap_hw.h [hunk 4]
```diff
@@ -16,7 +16,7 @@
 #include <stdint.h>
 
 /** Max number of tracking channels NAP configuration will be built with. */
-#define NAP_MAX_N_TRACK_CHANNELS     32
+#define NAP_MAX_N_TRACK_CHANNELS     MAX_CHANNELS
```
Comment (adel-mamin) [inline src/board/v3/nap/nap_hw.h]: Same here.

**Comment by pmiettinen** [comment]
https://github.com/swift-nav/libswiftnav-private/pull/25

**Comment by rge-exafore** [comment]
Closed as alternative implementation was merged (https://github.com/swift-nav/piksi_firmware/pull/719)

**Pull request closed without merging**